src/action/setting.js

```diff
@@ -16,13 +18,14 @@ class SettingAction {
   }
 
   setFiatCurrency({ fiat }) {
-    if (!FIATS[fiat]) {
-      throw new Error(`Invalid fiat currency: ${fiat}`);
-    }
-    this._store.settings.fiat = fiat;
+    this._store.settings.fiat = FIATS[fiat] ? fiat : DEFAULT_FIAT;
     this._wallet.getExchangeRate();
     this._db.save();
   }
+  setFiatByLocale({ locale }) {
+    const fiat = LocaleCurrency.getCurrency(locale).toLowerCase();
+    this.setFiatCurrency({ fiat });
+  }
```

Comment:
... `if (!FIATS[fiat])` falling back to `usd` if the app does not support the currency yet. This will allow us to detect Euro and Pounds for now and add more currency support in the future (which requires a design change in the settings).

Reply:
the ternary expression on line 21 does just that - if `fiat` is any of our supported fiats, it will be used. otherwise USD